### ISSUE TITLE
v3 collection methods

### ISSUE THREAD

**Issue by kcranston** [OpenTreeOfLife/germinator#71]
Looks like we did a bad job reviewing the [V3 spec for collections](https://docs.google.com/document/d/1XcgoQGJVxSgk-yKB-NAayQNZ2_QUQuw9_RIvbR2LeTw/edit#heading=h.yty047927s81). Some comments on the the methods listed:
* find_collections : when called without parameters, returns all collections. No effect of the verbose setting (i.e. always returns the entire collection blob for all collections)
* properties : not implemented (the [webapp does all filtering client-side](https://github.com/OpenTreeOfLife/opentree/blob/0340c0e7a35e527d13d95f529997cfc9f9f47a19/curator/static/js/tree-collection-dashboard.js))
* collection : works as described, e.g. `$ curl https://devapi.opentreeoflife.org/v3/collection/jimallman/asdf` 

I propose that we document only the existing functionality, or leave this undocumented for v3. Ideally for v4, it would be nice to implement find as GET with URL parameters (i.e. it seems silly to re-use the POST hacks that we have for studies with collections, since collections don't use the technology (neo4j) that makes these hacks necessary). 

Thoughts? @jar398 @jimallman

**Comment by jimallman** [comment]
Agreed that the [current implementation is half-baked](https://github.com/OpenTreeOfLife/phylesystem-api/blob/master/controllers/default.py#L222-L284). It could be improved to mimic the `properties` and `find_*` methods elsewhere, but my assumption was that collections would be added to a proper index (oti or its replacement) instead.

Perhaps the current `/v3/collections/find_collections` method could be renamed `/v3/collections/list_collections`? Either way, it's not going to follow the precedent of the `/v3/studies` methods unless I mimic them using the GitHub APIs. This would be a minor effort, perhaps a day to get all the response types and nuances correct?

**Comment by jimallman** [comment]
I _do_ think the existing methods are useful for API consumers, so I'd prefer to document this before too long. It's just a question of how much we intend to refine them, and whether API revisions will cause more annoyance than simply waiting until this is more mature.

**Comment by kcranston** [comment]
I say we leave it as find_collections but remove the documentation of parameter values. Then it is mostly consistent with find_studies and find_trees (save for the default value of 'verbose).

**Comment by kcranston** [comment]
The farther I get into the documentation, the more hesitant I am to include these in the v3 public methods. We just spent weeks changing all of the input and output parameter names for the taxonomy, tree_of_life and tnrs methods for consistency and clarity. The collection methods have not gone through the same scrutiny, and it seems highly probable that we will want to make changes to parameter names, formats, etc. We expect v4 to be coming shortly. So, we can:
* document 'as is' with the expectation that the next version will include many changes 
* review, and make changes for a v4 initial release

I am leaning towards the latter.

**Comment by jimallman** [comment]
> I am leaning towards the latter.

Yes, we can keep this "private" for now and help out users who need to understand it. 

Are you concerned that the JSON properties in the actual collections (`branch2sha`, `versionHistory`, etc) don't follow "v3 conventions"? These are mostly borrowed from Nexson, which is still a riot of unlike names.

**Comment by jar398** [comment]
At this point I'd say we don't need one more thing to review and document.
Let's leave it out of v3 until we can do it properly, but we can leave v2
as is - people can use both the v2 collections API and the v2 documentation
if they need to.

**Comment by jar398** [comment]
Seeing as my suggestion is not meeting a lot of friction, can we take that as a plan and close this issue? I'll assign to @kcranston who can make a decision and close it.
